client/src/VideosPerent.js

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useState } from "react";
+import VideoCard from "./VideoCard";
+
+export default function VideosPerent({ refreshVideos, setRefreshVideos }) {
+  const [videos, setVideos] = useState([]);
+  const [rating, setRating] = useState([]);
```

Comment:
Interesting - this `rating` state has a singular name (rating not ratings), but contains an array... And then when you use it, you iterate over each of the videos, so you're considering one particular video, but you pass the whole array of ratings as a prop to the `VideoCard`... This is a little surprising - I would generally expect the rating to be part of the video object (or, in unusual situations, maybe to have two arrays which have the same number of elements in them, and to iterate across them together in one loop), but this usage pattern is a little weird... Are you sure about it? How did you end up here? Happy to discuss! :)

Reply:
Doing coding in the middle of the night wasn't a great idea.
I rewrote everything on the next morning╰(*°▽°*)╯